steward/src/cellars/cellar.rs

```diff
@@ -0,0 +1,228 @@
+            };
+
+            Ok(CellarCalls::AddPosition(call).encode())
+        }
+        Function::PushPosition(params) => {
+            log_cellar_call(CELLAR_NAME, &PushPositionCall::function_name(), &cellar_id);
+            let call = PushPositionCall {
+                position: sp_call_parse_address(params.position)?,
+            };
+
+            Ok(CellarCalls::PushPosition(call).encode())
+        }
+        Function::RemovePosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemovePositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemovePositionCall {
+                index: params.index.into(),
+            };
+
+            Ok(CellarCalls::RemovePosition(call).encode())
+        }
+        Function::SetHoldingPosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetHoldingPositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetHoldingPositionCall {
+                new_holding_position: sp_call_parse_address(params.new_holding_position)?,
+            };
+
+            Ok(CellarCalls::SetHoldingPosition(call).encode())
+        }
+        Function::Rebalance(params) => {
+            log_cellar_call(CELLAR_NAME, &RebalanceCall::function_name(), &cellar_id);
+            let swap_params =
+                encode_swap_params(params.params.ok_or_else(|| {
+                    ErrorKind::SPCallError.context("swap params cannot be empty")
+                })?)?;
+
+            info!("encoded: {:?}", hex::encode(&swap_params));
+
+            let call = RebalanceCall {
+                from_position: sp_call_parse_address(params.from_position)?,
+                to_position: sp_call_parse_address(params.to_position)?,
+                assets_from: string_to_u256(params.assets_from)?,
+                exchange: (params.exchange - 1) as u8,
+                params: swap_params.into(),
+            };
+            let call = CellarCalls::Rebalance(call).encode();
+            info!("final call: {:?}", hex::encode(&call));
+            Ok(call)
+        }
+        Function::SetStrategistPayoutAddress(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetStrategistPayoutAddressCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetStrategistPayoutAddressCall {
+                payout: sp_call_parse_address(params.payout)?,
+            };
+
+            Ok(CellarCalls::SetStrategistPayoutAddress(call).encode())
+        }
+        Function::SetWithdrawType(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetWithdrawTypeCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetWithdrawTypeCall {
+                new_withdraw_type: params.new_withdraw_type as u8,
```

Comment:
We're setting an enum in the same fashion here as exhcange type above, except there's no `-1` even though the two enums in the contract have two entries and our enums have three entries respectively.

Reply:
Oof good catch. It needs the -1